airflow/serialization/serialized_objects.py

```diff
@@ -498,7 +498,7 @@ def serialize(
                 type_=DAT.SIMPLE_TASK_INSTANCE,
             )
         elif isinstance(var, Connection):
-            return cls._encode(var.to_dict(), type_=DAT.CONNECTION)
+            return cls._encode(var.to_json_dict(validate=True), type_=DAT.CONNECTION)
```

Comment:
Isn't that going to be backwards incompatible for older connections?

Reply:
It should be backward compatible.
If connection serialized by previous version, it should be deserialized by new one. I've also add test for that check.

I'm also not sure that we ever serialize connection in Airflow.

Reply:
Previous version serialize conn_id, description and uri new one every field except uri.

So this is basically two existed ways to create connection from constructor.